### PR TITLE
Added tests for SQLAlchemy validators, made model mandatory in AvailableAttr

### PR DESCRIPTION
**Pull request by shreyas-satish**

### CODE REVIEW

File: tests/fixtures.py [hunk 1]
```diff
@@ -5,6 +5,13 @@
 from baseframe import _, __
 import baseframe.forms as forms
 
+app1 = Flask(__name__)
+app1.config['SQLALCHEMY_DATABASE_URI'] = 'sqlite://'
+app1.config['SQLALCHEMY_TRACK_MODIFICATIONS'] = False
+app2 = Flask(__name__)
+app2.config['SQLALCHEMY_DATABASE_URI'] = 'postgresql:///baseframe_test'
```
Comment (jace) [inline tests/fixtures.py]: Three slashes?

File: tests/test_sqlalchemy.py [hunk 2]
```diff
@@ -0,0 +1,62 @@
...
+
+
+class ContainerForm(forms.Form):
+    name = forms.StringField(__("Name"), validators=[forms_sqlachemy.AvailableName()])
+    title = forms.StringField(__("Title"), validators=[forms_sqlachemy.AvailableAttr('title')])
+    content = forms.TextAreaField(__("Content"))
+
+
+class FormSQLAlchemyTestCase(BaseframeTestCase):
+    def setUp(self):
```
Comment (jace) [inline tests/test_sqlalchemy.py]: Add `app = app1` here and use `self.app` in `setUp`. You can make a subclass below that substitutes with `app = app2` to re-run tests on the other database.

Comment (jace) [inline tests/test_sqlalchemy.py]: Don't inherit from `BaseframeTestCase` as that messes with `self.app`. Just add the `init_app` call here.

File: tests/fixtures.py (hunk 1 above)
Comment (shreyas-satish) [inline tests/fixtures.py]: My development config for PostgreSQL is specified slightly differently (with the username and port). So I borrowed the config from [coaster](https://github.com/hasgeek/coaster/blob/master/tests/test_models.py#L26).

Comment (shreyas-satish) [inline tests/fixtures.py]: Three slashes seems to be the correct syntax, two slashes errors out.

File: baseframe/forms/sqlalchemy.py [hunk 3]
```diff
@@ -25,6 +25,8 @@ def __init__(self, attr, message=None, model=None):
 
     def __call__(self, form, field):
         model = self.model or form.edit_model
+        if not model:
+            raise TypeError(u"Either the field or the form MUST be linked to a model")
```
Comment (jace) [inline baseframe/forms/sqlalchemy.py]: `self` is the validator, not the field. Either the validator or the form must be linked to a model.

**Pull request merged**